doc/spark_on_ray.md

```diff
@@ -1,3 +1,26 @@
+### How to schedule Spark Master to a designated node
```

Comment:
What do you think about `Spark master node affinity`? I think this allows you to send drivers to a set of nodes, not only one.

Reply:
In fact, I am thinking maybe we should do this for executors as well, do different job's executor can land on different worker nodes. This is a unique feature for Spark on Ray and it is useful for people to use one cluster host multiple jobs with different resource requirement.

Reply:
> What do you think about Spark master node affinity? I think this allows you to send drivers to a set of nodes, not only one.

Hmm, I was thinking as a normal user, who just cares about one application at a time. But yes, your suggested name is also fine.

Doing this for executors is trickier, because each executor can require different resource? Or we limit it to same resource for every executor.

Reply:
> Hmm, I was thinking as a normal user, who just cares about one application at a time.

Yeah in that case the user might not even need this PR because most likely his/her driver node will be run on headnode. But when the Ray cluster is crowded, it is a different story :)

> Doing this for executors is trickier, because each executor can require different resource? Or we limit it to same resource for every executor.

So in our usecase we have all the different jobs belong to one batch automation pipeline usecase all run in one Ray cluster, in this case we save the cluster launch/tear down time. For executor, what we need is give use a custom config to assign extra custom resources, so the user can assign the executors for one particular job to a node config (and one Ray cluster might have multiple node configs for different types of wirker nodes).

Reply:
Sure, let's do it in another pr.